echo/echo_test.go

```diff
@@ -0,0 +1,78 @@
+package echo
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdater_Update(t *testing.T) {
+	tests := []struct {
+		name          string
+		testFile      string
+		wantErr       bool
+		expectedFiles map[string]string
+	}{
+		{
+			name:     "valid response",
+			testFile: "testdata/valid.json",
+			expectedFiles: map[string]string{
+				"nginx.json":  "testdata/golden/nginx.json",
+				"python.json": "testdata/golden/python.json",
+				"redis.json":  "testdata/golden/redis.json",
+			},
+		},
+		{
+			name:     "invalid JSON response",
+			testFile: "testdata/invalid.json",
+			wantErr:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if tt.testFile != "" {
+					http.ServeFile(w, r, tt.testFile)
+				} else {
+					w.WriteHeader(http.StatusInternalServerError)
```

Comment:
I think we can add testcase for this case.

Reply:
Done 👍